src/include/sof/drivers/interrupt.h

```diff
@@ -36,18 +36,23 @@ struct irq_child {
 	struct list_item list;
 };
 
+/**
```

Comment:
Doxygen standard goes like this:
```
/** \brief Interrupt client descriptor. */
struct irq_desc {
        int irq;                                   /**< virtual IRQ number */
        void (*handler)(void *arg);   /**< interrupt handler function */
        ...
};
```

Reply:
@tlauda indeed, the style I used is from the Linux kernel, it isn't the classic doxygen style. It looks like the "pure" doxygen doesn't define a way to document struct / class members in a header, not inline. So, we'd end up with
```
/**
 * \brief cascading interrupt controller descriptor.
 */
struct irq_cascade_desc {
	const char *name;				/**< name of the
							 * controller
							 */
	const struct irq_cascade_ops *ops;		/**< cascading interrupt
							 * controller driver
							 * operations
							 */
	struct irq_desc desc;				/**< the interrupt, that
							 * this controller is
							 * generating
							 */
	struct irq_cascade_desc *next;			/**< link to the global
							 * list of interrupt
							 * controllers
							 */
	bool global_mask;				/**< the controller
							 * cannot mask input
							 * interrupts per core
							 */
	spinlock_t lock;				/**< protect child
							 * lists, enable and
							 * child counters
							 */
	int enable_count[PLATFORM_CORE_COUNT];		/**< enabled child
							 * interrupt counter
							 */
	unsigned int num_children[PLATFORM_CORE_COUNT];	/**< number of children
							 */
	struct irq_child child[PLATFORM_IRQ_CHILDREN];	/**< array of child
							 * lists - one per
							 * multiplexed IRQ
							 */
};
```
Is this ok?